zeq/hbp/vocabulary.h

```diff
@@ -23,9 +37,46 @@ static const uint128_t EVENT_SELECTED_IDS(
     lunchbox::make_uint128( "hbp::SelectedIDsEvent" ));
 static const uint128_t EVENT_TOGGLE_ID_REQUEST(
     lunchbox::make_uint128( "hbp::ToggleIDRequest" ));
+static const lunchbox::uint128_t EVENT_REQUEST(
+    lunchbox::make_uint128( "hbp::RequestEvent" ));
+static const lunchbox::uint128_t EVENT_IMAGE_RAW_RGBA8(
+    lunchbox::make_uint128( "hbp::imageRawRGBA8Event" ));
 //@}
 
 /**
+ * Serialize the given rgba raw image into an Event of type EVENT_IMAGE_RAW.
+ * The image format must be raw rgba with 8bits per component.
+ * @param image the image.
+ * @return the serialized event.
+ */
+::zeq::Event serializeimageRawRGBA8( const ImageRawRGBA8& image );
```

Comment:
Knowing the data image data is passed by shallow pointer copy I'd like to see either:
- A constructor for ImageRawRGA8 that takes the resolution and data as parameters
- or a version of serialize**I**mageRawRGBA (uppercase I) that takes those parameters directly.

Personally, I prefer the second one, otherwise the comment about memory management in lines 22-23 is not precise because it says nothing about where the memory comes from in this case.